server/heroku/heroku.go

```diff
@@ -141,3 +143,24 @@ func RangeHeader(r *http.Request) (headerutil.Range, error) {
 	}
 	return *rangeHeader, nil
 }
+
+// key used to store context values from within this package.
+type key int
+
+const (
+	userKey key = 0
+)
+
+// WithUser adds a user to the context.Context.
+func WithUser(ctx context.Context, u *empire.User) context.Context {
```

Comment:
Moving these functions from empire core to here so that they can't be used within empire core. Providing the user that's performing an action should happen in the options struct provided.